swap_meet/vendor.py

```diff
@@ -1,2 +1,148 @@
+                return i
+
+        return False
+
+    # ********* Wave 2 **********
+
+    # this function returns a list of items with same category
+    def get_by_category(self, category):
+        same_category = []
+        for item in self.inventory:  # item is an instance of the Item
+            if item.category == category:
+                same_category.append(item)
+
+        return same_category
+
+    # ********* Wave 3 **********
+
+    def swap_items(self, friend_vendor, my_item, friend_item):
+
+        """
+        This function takes in 3 arguments:
+            friend_vendor - Vendor's class object
+            my_item - Item's class object
+            friend_item - Item's class object
+        """
+
+        # if neither vendor or friend_vendor contain the item return false
+        if not self.contains(my_item) or not friend_vendor.contains(friend_item):
+            return False
+
+        # remove my_item and append to friend_vendor
+        self.remove(my_item)
+        friend_vendor.add(my_item)
+
+        # remove friend_item from list and append to inventory
+        friend_vendor.remove(friend_item)
+        self.add(friend_item)
+
+        return True
+
+    # this function checks to see if item is in list, return boolean
+    def contains(self, item):
+        return item in self.inventory
+
+    # ********* Wave 4 **********
+
+    def swap_first_item(self, friend_vendor):
+
+        """
+        -Remove first item from inventory and append to friend's inventory
+        -Remove friend's first item and append to vendor's inventory
+        -If either list is empty return False
+        """
+        
+        if len(self.inventory) == 0 or len(friend_vendor.inventory) == 0:
+            return False
+
+        # pop returns the item that it removed
+        vendor_first_item = self.inventory.pop(0)
+        friend_vendor.inventory.append(vendor_first_item)
+
+        friend_first_item = friend_vendor.inventory.pop(0)
+        self.inventory.append(friend_first_item)
+
+        return True
+
+    # ********* Wave 6 **********
+
+    def get_best_by_category(self, item_category):
+        """
+        -item_category: a string that represents a category
+        -get item with best conditions
+        -checking for highest condtion and matching category and return item
+        -if no item matches category return None
+        -return single item even if there are duplicates items & conditions
+        """
+
+        result = []
+        for item in self.inventory:
+            if item.category == item_category:
+                result.append(item)
+
+        highest_condition = -1
+        highest_condition_item = None
+        for item in result:
+            if item.condition > highest_condition:
+                highest_condition = item.condition
+                highest_condition_item = item
+
+        return highest_condition_item
+
+    def swap_best_by_category(self, other, my_priority, their_priority):
```

Comment:
This function is really clean due to utilizing helper functions! Great work Zandra!